src/main/java/com/fasterxml/jackson/databind/deser/std/FactoryBasedEnumDeserializer.java

```diff
@@ -132,18 +135,21 @@ public Object deserialize(JsonParser p, DeserializationContext ctxt) throws IOEx
             // 30-Mar-2020, tatu: For properties-based one, MUST get JSON Object (before
             //   2.11, was just assuming match)
             if (_creatorProps != null) {
-                if (!p.isExpectedStartObjectToken()) {
+                if (p.isExpectedStartObjectToken()) {
+                    _propCreatorRef.compareAndSet(null,
```

Comment:
But wouldn't this create (and try set) a new instance every time? :-D
(instead of trying to `get()` and only create-and-set if `null`)

Reply:
No, my problem is that instead of only creating it if needed, it creates new instance every time and sets -- and never uses value. You are correct in that get + set is not atomic, but I am not worried about atomicity here; if instance gets created 2 or 3 times that's fine. They are all identical. Goal is just to avoid always creating instance eagerly.